src/field/gen_fieldops.jinja

```diff
@@ -90,16 +100,7 @@ void autogen_{{lhs}}_{{rhs}}_{{operator_name}}(
     {% endif %}
     );
 
-    {% if lhs == rhs == "Field3D" %}
-#if CHECK > 0
-    if (this->getLocation() != rhs.getLocation()) {
-      throw BoutException(
-        "Error in {{operator_name}}: fields at different locations. lhs is at %s, rhs is at %s!",
-        strLocation(this->getLocation()), strLocation(rhs.getLocation()));
-    }
-#endif
     checkData(*this);
```

Comment:
Just to check -- this was previously inside the `if lhs == rhs == "Field3D"` template block but no longer is, is this an intentional change alongside moving the CHECK block? I realise it only makes sense to check data on `*this`  after the operation, but do we still need to ensure that we only do the check if we're working with Field3D values?

Reply:
There is now a `checkData` overload for BoutReal/Field2D/Field3D, so we should just be able to call this unconditionally.